// react-native/Libraries/NewAppScreen/components/Header.js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';
import Colors from './Colors';
import type {Node} from 'react';
import {Text, StyleSheet, ImageBackground} from 'react-native';
import React from 'react';

const Header = (): Node => (
  <ImageBackground
    accessibilityRole={'image'}
    source={require('./logo.gif')}
    style={styles.background}
    imageStyle={styles.logo}>
   
  </ImageBackground>
);

const styles = StyleSheet.create({
  background: {
   paddingBottom: 100,
    paddingTop: 100,
    paddingHorizontal: 40,
    backgroundColor: Colors.lighter,
    width:190,
    height:200,
  },
  logo: {
    opacity: 1.0,
    overflow: 'visible',
    resizeMode: 'cover',
    /*
     * These negative margins allow the image to be offset similarly across screen sizes and component sizes.
     *
     * The source logo.png image is 512x512px, so as such, these margins attempt to be relative to the
     * source image's size.
    */
    marginLeft: 65, 
    marginRight: 65,
    
  },
  text: {
    fontSize: 40,
    fontWeight: '600',
    textAlign: 'center',
    color: Colors.black,
  },
});

export default Header;
