// react-native/Libraries/NewAppScreen/components/LearnMoreLinks.js
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';
import Colors from './Colors';
import type {Node} from 'react';
import openURLInBrowser from 'react-native/Libraries/Core/Devtools/openURLInBrowser';
import {View, Text, StyleSheet, TouchableOpacity, ScrollView, ImageBackground, Image} from 'react-native';

import React from 'react';

const links = [
  {
    id: 1,
    category: 'Advertisement',
    title: 'Please subscribe to view \n more exclusive contents.', 
    image: require('./assets/images/localAd-1.png'),
    link: 'https://www.u.com.my/',
    
  },
  {
   id: 2,
    category: 'Advertisement',
    title: '2 months for only Rm15.90 \n        for offline movies.', 
    image: require('./assets/images/LocalAd-2.jpg'),
    link: 'https://www.gsc.com.my/',
  },
  {
     id: 3,
    category: 'SCIENCE FICTION',
    title: 'Blade Runner Future 2049.',
    image: { uri: 'https://i.imgur.com/L68FtMA.jpg' }, 
    starimage: require('./StarRating/4star.png'),
    rating: 4.0,
    link: 'https://www.gsc.com.my/main/home',
    
  },
  {
    id: 4,
    category: 'SUPERHERO ACTION',
    title: 'Captain America Civil War.',
    image: { uri: 'https://i.imgur.com/tAui2H7.jpg'}, 
    starimage: require('./StarRating/4star.png'),
    rating: 4.0,
    link: 'https://www.gsc.com.my/main/home',
    
  },
  {
    id: 5,
    category: 'ROMANCE',
    title: 'Beauty and the Beast.',
    image: { uri: 'https://images-na.ssl-images-amazon.com/images/I/91JHbBQPriL.jpg'}, 
    starimage: require('./StarRating/2star.png'),
    rating: 2.5,
    link: 'https://www.gsc.com.my/main/home',
  
  },
  {
    id: 6,
category: 'ACTION',
title: 'Sonic The Hedgehog',
image: { uri: 'https://cdn.collider.com/wp-content/uploads/2020/01/sonic-the-hedgehog-poster.jpeg' }, 
rating: 3.8,
starimage: require('./StarRating/3star.png'),
    link: 'https://www.gsc.com.my/main/home',
  },   
  {

    id: 7,
category: 'ROMANCE',
title: 'Shawshank Redemption.',
image: { uri: 'https://images-na.ssl-images-amazon.com/images/I/911USrdQtPL._AC_SY679_.jpg' }, 
starimage: require('./StarRating/3star.png'),
rating: 3.0,
    link: 'https://www.gsc.com.my/main/home',
    },
  {
   id: 8,
category: 'SCIENCE FICTION',
title: 'TERMINATOR : Dark Fate',
image: { uri: 'https://images-na.ssl-images-amazon.com/images/I/71mUOdelVeL._AC_SY741_.jpg' }, 
starimage: require('./StarRating/4star.png'),
rating: 4.0,
    link: 'https://www.gsc.com.my/main/home',
    
  },
];

const LinkList = (): Node => (
  <ScrollView horizontal={true} style={styles.container}>
    {links.map(({id, title, link, description, image, rating, category, starimage}) => {
      return (
        <React.Fragment key={id}>
          <View style={styles.separator} />
          <TouchableOpacity
            accessibilityRole={'button'}
            onPress={() => openURLInBrowser(link)}
            style={styles.linkContainer}>
            <Image style={styles.theImage} source={image}/>
             <Text style={styles.category}>{category}</Text>
            <Text style={styles.link}>{title}</Text> 
         <Image style={styles.starimage} source={starimage}/>

           
          </TouchableOpacity>
        </React.Fragment>
      );
    })}
  </ScrollView>
);

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    paddingHorizontal: 14,
   

  },
  linkContainer: {
    flexWrap: 'wrap',
    flexDirection: 'column',
    justifyContent: 'space-between',
    backgroundColor: '#1D1D1D',
    alignItems: 'center',
    paddingVertical: 1,
    margin:0,
  },
  link: {
    flex: 3,
    fontSize: 18,
    fontWeight: '500',
    color: Colors.light,
    justifyContent: 'center',
  },
  category: {
    flex: 3,
    fontWeight: 'bold',
    fontStyle: 'italic',
    fontSize: 20,
  
    color: 'yellow',
  },
  title: {
    flex: 3,
    paddingVertical: 16,
    fontWeight: '500',
    fontSize: 18,
    color: Colors.light,
  },
  description: {
    flex: 3,
    paddingVertical: 16,
    fontWeight: '500',
    fontSize: 19,
    color: Colors.light,
  },
  separator: {
    backgroundColor: Colors.light,
    height: 1,
  },
  theImage: {
        flex: 1,
    height: 200,
    width:200,
    backgroundColor: "black",
    margin: 7.5,
    justifyContent: "center",
    alignItems: "center"
    },

 starimage: {
  width: 100,
  height: 25,
  margin:10,
 },
});

export default LinkList;
